posthog/models/surveys/survey.py

```diff
@@ -273,6 +277,49 @@ def get_file_system_representation(self) -> FileSystemRepresentation:
             should_delete=False,
         )
 
+    def scheduled_changes_dispatcher(
+        self, payload, user: Optional[AbstractBaseUser] = None, scheduled_change_id: Optional[int] = None
+    ):
+        from posthog.api.survey import SurveySerializer
+
+        if "launch" not in payload and "end" not in payload:
+            raise Exception("Payload must contain either 'launch' or 'end' key")
+
+        # Store scheduled change context on the instance for activity logging
+        if scheduled_change_id is not None:
+            self._scheduled_change_context = {"scheduled_change_id": scheduled_change_id}
+
+        http_request = HttpRequest()
+        # We kind of cheat here and set the request user to the user who created the scheduled change
+        # It's not the correct type, but it matches enough to get the job done
+        http_request.user = user or self.created_by  # type: ignore
+        http_request.method = "PATCH"  # This is a partial update, not a new creation
+        context = {
+            "request": http_request,
+            "team_id": self.team_id,
+            "project_id": self.team.project_id,
+        }
+
+        serializer_data = {}
+        if payload.get("launch"):
+            # this survey is already running, nothing to do here.
+            # do we want to restart the survey if its already ended?
+            if self.start_date and not self.end_date:
+                return
+
+            serializer_data["start_date"] = datetime.now()
```

Comment:
I'd like to add analytics to understand if this feature is successful. I'm not 100% sure where that should go.  One idea is to add analytics here to emit events if a survey is started via a schedule. Pointers are welcome :)